hopp/simulation/technologies/wind/wind_plant.py

```diff
@@ -191,17 +199,68 @@ def __attrs_post_init__(self):
 
         self._dispatch = None
 
-        self.turb_rating = self.config.turbine_rating_kw
+        if self.config.turbine_rating_kw is not None:
+            self.turb_rating = self.config.turbine_rating_kw
         self.num_turbines = self.config.num_turbines
+            
+        if self.config.model_name=="pysam":
+            self.initialize_pysam_wind_turbine()
+        
+        
+    def initialize_pysam_wind_turbine(self):
+        """Initialize wind turbine parameters for PySAM simulation.
+
+        Raises:
+            UserWarning: discrepancy in rotor_diameter value
+            UserWarning: discrepancy in hub-height value
+        """
+
+        if self.config.model_name != "pysam":
+            return
+        
         if self.config.rotor_diameter is not None:
             self.rotor_diameter = self.config.rotor_diameter
-
-        if self.config.model_name=="pysam":
-            if self.config.hub_height is not None:
-                self._system_model.Turbine.wind_turbine_hub_ht = self.config.hub_height
-            if self.config.adjust_air_density_for_elevation and self.site.elev is not None:
-                air_dens_losses = calculate_air_density_losses(self.site.elev)
-                self._system_model.Losses.assign({"turb_specific_loss":air_dens_losses})
+        
+        if self.config.turbine_name is not None:
+            valid_name = check_turbine_library_for_turbine(self.config.turbine_name)
+            if not valid_name:
+                turbine_name = check_turbine_name(self.config.turbine_name)
+                logger.warning(f"closest matching turbine name to {self.config.turbine_name} is {turbine_name} ... setting turbine model as {turbine_name}")
+            else:
+                turbine_name = self.config.turbine_name
+            turbine_dict = get_pysam_turbine_specs(turbine_name,self)
+            self._system_model.Turbine.assign(turbine_dict)
+            self.rotor_diameter = turbine_dict["wind_turbine_rotor_diameter"]
+            self.turb_rating = max(turbine_dict["wind_turbine_powercurve_powerout"])
+
+            if self.config.rotor_diameter is not None:
+                if self.config.rotor_diameter != self._system_model.Turbine.wind_turbine_rotor_diameter:
+                    raise UserWarning(f"input rotor diameter ({self.config.rotor_diameter}) does not match rotor diameter for turbine ({self._system_model.Turbine.wind_turbine_rotor_diameter})")
```

Comment:
Can you split this across multiple lines? I'm not aiming for a specific line length but once it gets past well beyond 122 I think it's safe to break up as a good practice. There are a few more here that would be nice as well. I recommend the below form.

```python
msg = (
    "message"
    " continued"
)
raise UserWarning(msg)
```